src/Components/Components/src/Reflection/ComponentResolver.cs

```diff
@@ -6,28 +6,45 @@
 using System.Linq;
 using System.Reflection;
 
-namespace Microsoft.AspNetCore.Components
+namespace Microsoft.AspNetCore.Components.Reflection
 {
     /// <summary>
     /// Resolves components for an application.
     /// </summary>
-    internal static class ComponentResolver
+    public sealed class ComponentResolver
```

Comment:
uhhhhhhhhhhhhhhhhhhhhhhhhhhhh

Reply:
I can make an interface. I really want to avoid IVTs to things from assemblies in the shared fx to outside.

Reply:
Why do we need an interface? What's the requirement to make this public?

Reply:
We need to be able to register it in DI. Another alternative is to expose an API from Components that registers services and have WebAsm and ServerSide call that as part of their DI set up

Reply:
Sorry, I still don't understand this!

 * What's insufficient about the previous design where `ResolveComponents` was `static`? Can't we also have a `static Dictionary<Assembly, IReadOnlyList<Type>>` to cache the results of `ResolveComponents` calls?
 * Why would we only cache the output from the *first* calls to `ResolveComponents` (when `_cachedAssembly == null`)? Why not cache all the results?